src/esp.d.ts

```diff
@@ -0,0 +1,83 @@
+// notice_start
+/*
+ * Copyright 2015 Dev Shop Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// notice_end
+
+declare module esp {
+
+    function observeEvent(eventName:String, observationStage?:string): any;
+
+    class Router {
+        addModel<T>(modelId:String, model:T, options) : void;
+        removeModel(modelId:String) : void;
+        publishEvent(modelId:String, eventType:String, event:any) : void;
+        broadcastEvent(eventType:String, event:any)  : void;
+        executeEvent(eventType:String, event:any)  : void;
+        runAction(modelId:String, action:() => void)  : void;

```

Comment:
We would just do an overload.

On 15 March 2016 at 15:43, Keith Woods <notifications@github.com> wrote:

> In src/esp.d.ts
> <https://github.com/esp/esp-js/pull/82#discussion_r56185955>:
>
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +// notice_end
> > +
> > +declare module esp {
> > +
> > +    function observeEvent(eventName:String, observationStage?:string): any;
> > +
> > +    class Router {
> > +        addModel<T>(modelId:String, model:T, options) : void;
> > +        removeModel(modelId:String) : void;
> > +        publishEvent(modelId:String, eventType:String, event:any) : void;
> > +        broadcastEvent(eventType:String, event:any)  : void;
> > +        executeEvent(eventType:String, event:any)  : void;
> > +        runAction(modelId:String, action:() => void)  : void;
>
> the action gets passed the model, but often you don't need to use it. Not
> sure how'd you'd represent that. As an overload?
> runAction<TModel>(modelId:String, action:(model:TModel) => void) : void;
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/esp/esp-js/pull/82/files#r56185955
>